server/src/main/java/org/elasticsearch/search/aggregations/bucket/range/InternalRange.java

```diff
@@ -265,11 +265,11 @@ public InternalRange(StreamInput in) throws IOException {
         for (int i = 0; i < size; i++) {
             String key = in.getVersion().onOrAfter(Version.V_7_17_1) ? in.readOptionalString() : in.readString();
             double from = in.readDouble();
-            if (in.getVersion().onOrAfter(Version.V_7_17_0)) {
+            if (in.getVersion().onOrAfter(Version.V_7_17_0) && in.getVersion().before(Version.V_8_2_0)) {
                 in.readOptionalDouble();
```

Comment:
As a result of #83175 here we need to overwrite both `from` and `to` with whatever value we read from the wire if version matches `7.17.0`. Version `7.17.0` indeed, writes `from`, `originalFrom`, `to`, `originalTo` on the wire. In that case, if we just drop `originalFrom` and `originalTo` we have incorrect results because we would use the `from` and `to` during serialisation, which have precision issues.